modules/diff.js

```diff
@@ -152,27 +149,48 @@ const getClientMeta = async (product) => {
   return { head, base, dev, prod }
 }
 
+const worker = async ({ product, response_url }) => {
+  let r = {
+    response_type: 'ephemeral',
+    text: `Sorry, product ${product} not found`,
+  }
 
-router.all('/', verifySlack, async (req, res, next) => {
-  const { text: product = 'firstorder' } = req.body || {}
+  if (Object.keys(SERVICES).includes(product)) {
+    r = await getGitDiff({ product, ...await getServiceMeta(product) })
+  } else if (Object.keys(CLIENTS).includes(product)) {
+    r = await getGitDiff({ product, ...await getClientMeta(product) })
+  }
 
-  try {
-    let r = {
-      response_type: 'ephemeral',
-      text: `Sorry, product ${product} not found`,
-    }
+  return axios.post(response_url, { replace_original: true, ...r })
+}
 
-    if (Object.keys(SERVICES).includes(product)) {
-      r = await getGitDiff({ product, ...await getServiceMeta(product) })
-    } else if (Object.keys(CLIENTS).includes(product)) {
-      r = await getGitDiff({ product, ...await getClientMeta(product) })
-    }
+const route = (req, res) => {
+  const { text: product = 'firstorder', response_url } = req.body || {}
 
-    return res.status(200).json(r)
-  } catch(err) {
-    return next(err)
-  }
-})
+  const payload = { product, response_url }
 
+  if (DEPLOYED) {
+    lambda.invoke({
+      FunctionName: getFuncName('diff'),
+      InvocationType: 'Event',
+      Payload: JSON.stringify(payload),
+    }, (err) => {
+      if (err) {
+        console.error(err)
+        return res.status(200).json({ response_type: 'ephemeral', text: 'Failed to diff' })
+      }
+      return res.status(200).json({
+        response_type: 'ephemeral',
+        text: `Diffing for ${product}...`,
+      })
+    })
+  } else {
+    worker(payload).catch(console.error)
```

Comment:
this mimics the immediate response + worker response, right?

Reply:
yea, so `worker` is `async`, this works for local testing setup (or perpetually running server setup)